CardsCashCasino/Manager/TexasHoldEmManager.cs

```diff
@@ -810,199 +813,249 @@ private static void OnTimeoutEvent(object source, ElapsedEventArgs e)
-            public void AddToPot(int _currentBet, List<int> playerBets, List<int> playersActive)
+            /// <param name="value">The amount to decrement the pot's total by.</param>
+            public void DecrementPot(int value)
             {
-                if (playerBets.All(n => n == playerBets[0]))
+                if (value <= 0)
                 {
-                    Pots.First(pot => pot.PotType == PotType.MAIN).Total += _currentBet * playerBets.Count();
+                    throw new ArgumentException("Decrement value must be positive.");
                 }
-                else
+
+                if (Total - value < 0)
                 {
-                    CreateSidePots(_currentBet, playerBets, playersActive);
+                    throw new InvalidOperationException("Pot total cannot be negative.");
                 }
+
+                Total -= value;
             }
+
             /// <summary>
-            //Adds bets of players that were folded to the pot
+            /// Decrement the total of the pot by a specified value.
             /// </summary>
-            /// <param name="playersBets"> list of bets that were placed but folded due to a raise in that round</param>
-            public void AddFoldedBetsToPot(List<int> playerBets)
+            /// <param name="playerIndex">The position of the player being removed from pot eligiblity .</param>
+            public void RemoveEligiblePlayer(int playerIndex)
             {
-                if (playerBets.Count() == 0)
+                if (EligiblePlayers.Contains(playerIndex))
                 {
-                    return;
-                }
-                else
-                {
-                    for (int bet = 0; bet < playerBets.Count(); bet++)
-                    {
-                        Pots[0].Total += playerBets[bet];
-                    }
+                    EligiblePlayers.Remove(playerIndex);
                 }
             }
-            /// <summary>
-            /// Creates side pots as needed at the end of a round of betting.
-            /// </summary>
-            /// <param name="_currentBet">The amount to add to the pot.</param>
-            /// <param name="playerBets">The list of bets from players still active after the round of betting.</param>
-            /// <param name="playersActive">The list of players eligible to win the main pot, all in players will be subtracted for side pot</param>
-            public void CreateSidePots(int _currentBet, List<int> playerBets, List<int> playersActive)
+        }
+    }
+
+    public class TexasHoldEmPotManager
+    {
+        // List to hold the pots for the game
+        public List<TexasHoldEmManager.TexasHoldEmPot> Pots { get; set; }
+
+        public TexasHoldEmPotManager()
+        {
+            Pots = new List<TexasHoldEmManager.TexasHoldEmPot>(); // Initialize the Pots list
```

Comment:
this does not need to be initalized in the constructor. just initialize it where it is created above, and remove the constructor.